Sources/MultiSlider.swift

```diff
@@ -157,7 +157,11 @@ open class MultiSlider: UIControl {
             let location = panGesture.location(in: slideView)
             draggedThumbIndex = closestThumb(point: location)
         case .ended, .cancelled, .failed:
-            sendActions(for: .touchDragExit)
+            if bounds.contains(panGesture.location(in: self)) {
+                sendActions(for: .touchUpInside)
+            } else {
+                sendActions(for: .touchUpOutside)
```

Comment:
When I test `UISlider` I only get `.touchUpInside`, never outside. So no need for the if-else.

Reply:
I just tested it myself: I got `.touchUpInside` when I let go inside the `UISlider` and `.touchUpOutside` when I let go outside the control.

I tested it with a new `UISlider` in the MultiSliderDemo project adding targets for `.touchUpInside` and `.touchUpOutside`.

Of course I can change the PR to only send `.touchUpInside` (even when the drag ends outside) but it would be inconsistent with at least some versions of `UISlider` and also less precise.

I vote to add inside and outside actions :)

Reply:
Attached is the project I used for testing. I can only get `.touchUpOutside` to fire if I drag very far outside the slider, not merely out of it's frame.

[SliderEvents.zip](https://github.com/yonat/MultiSlider/files/2150619/SliderEvents.zip)

Reply:
Got it! Somehow there seems to be an extended bounding rect around the `UISlider`. To get the `.touchUpOutside` event you have to drag far outside the control (until `.touchDragExit` is fired). So in principle `UISlider` really does differ between inside and outside for touchUp, although it defines ‘outside’ in a funny and undocumented(?) way.

So what do you think of leaving it like it is (as the logic for extending the testing bounds for me is questionable)?

Alternatively we could define a testing bounds extension ourselves:

``` swift
private let boundsExtensionForTouchUpOutsideEvents: CGFloat = 30
// […]
let testBounds = bounds.insetBy(dx: -boundsExtensionForTouchUpOutsideEvents, dy: -boundsExtensionForTouchUpOutsideEvents)
if testBounds.contains(panGesture.location(in: self)) {
    sendActions(for: .touchUpInside)
} else {
    sendActions(for: .touchUpOutside)
}
```

…and the last option I see would be to only send .touchUpInside, which breaks consistency with `UISlider` the most.